modules/integration/tests-iot-web-ui/src/test/java/org/wso2/carbon/iot/integration/web/ui/test/extension/CustomTestServerManager.java

```diff
@@ -0,0 +1,147 @@
+        } else {
+            throw new IllegalArgumentException("portOffset value must be set in command list");
+        }
+        this.commandMap = commandMap;
+    }
+
+    public String getCarbonZip() {
+        return carbonZip;
+    }
+
+    public String getCarbonHome() {
+        return carbonHome;
+    }
+
+    public int getPortOffset() {
+        return portOffset;
+    }
+
+    public void configureServer() throws AutomationFrameworkException {
+
+    }
+
+
+    public Map<String, String> getCommands() {
+        return commandMap;
+    }
+
+    /**
+     * This method is called for starting a Carbon server in preparation for execution of a
+     * TestSuite
+     * <p/>
+     * Add the @BeforeSuite TestNG annotation in the method overriding this method
+     * @param server : The server which needs to be start.
+     * @return The CARBON_HOME
+     * @throws java.io.IOException If an error occurs while copying the deployment artifacts into the
+     *                             Carbon server
+     */
+    public String startServer(String server)
+            throws AutomationFrameworkException, IOException, XPathExpressionException {
+        if(carbonHome == null) {
+            if (carbonZip == null) {
+                carbonZip = System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_CARBON_ZIP_LOCATION);
+            }
+            if (carbonZip == null) {
+                throw new IllegalArgumentException("carbon zip file cannot find in the given location");
+            }
+            carbonHome = carbonServer.setUpCarbonHome(carbonZip) + "/" + server;
+            configureServer();
+        }
+        log.info("Carbon Home - " + carbonHome );
+        if (commandMap.get(ExtensionConstants.SERVER_STARTUP_PORT_OFFSET_COMMAND) != null) {
+            this.portOffset = Integer.parseInt(commandMap.get(ExtensionConstants.SERVER_STARTUP_PORT_OFFSET_COMMAND));
+        } else {
+            this.portOffset = 0;
+        }
+        carbonServer.startServerUsingCarbonHome(carbonHome, commandMap);
+        return carbonHome;
+    }
+
+    /**
+     * Restarting server already started by the method startServer
+     * @throws AutomationFrameworkException
+     */
+    public void restartGracefully() throws AutomationFrameworkException {
+        if(carbonHome == null) {
+            throw new AutomationFrameworkException("No Running Server found to restart. " +
+                    "Please make sure whether server is started");
+        }
+        carbonServer.restartGracefully();
+    }
+
+    /**
+     * This method is called for stopping a Carbon server
+     * <p/>
+     * Add the @AfterSuite annotation in the method overriding this method
+     *
+     * @throws AutomationFrameworkException If an error occurs while shutting down the server
+     */
+    public void stopServer() throws AutomationFrameworkException {
+        carbonServer.serverShutdown(portOffset);
+    }
+
```

Comment:
extra lines

Reply:
This is fixed in https://github.com/wso2/product-iots/pull/1454/commits/caba24a1e13612fd30c8e2befa5c5d713a64dee8